Wangscape/AlphaCalculator.cpp

```diff
@@ -0,0 +1,85 @@
+#include "AlphaCalculator.h"
+#include <algorithm>
+#include <numeric>
+
+
+
+AlphaCalculator::AlphaCalculator()
+{
+}
+
+
+AlphaCalculator::~AlphaCalculator()
+{
+}
+
+void AlphaCalculator::calculatePixelAlphaLinear(WeightVector& weights, AlphaVector& alphas)
+{
+    int err = 255;
+    float total_weight = 0.f;
+    for (auto& weight : weights)
+    {
+        weight = std::max(0.f, weight);
+        total_weight += weight;
+    }
+    if (total_weight <= 0.f)
+    {
+        // This shouldn't happen, but it is easy to make weight masks
+        // where this does happen.
+        for (auto& alpha : alphas)
+        {
+            alpha = 0;
+        }
+        return;
+    }
+    for (size_t i = 0; i < weights.size(); i++)
+    {
+        int alpha = (int)((255 * weights[i]) / total_weight);
```

Comment:
`total_weight <= 0.f` will catch `total_weight == 0`, but there is still danger of overflow when it is positive but very small. The comparison should be made to a very small number (1000*epsilon would probably be enough) to catch that case.